src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigToJobMapper.java

```diff
@@ -141,89 +143,107 @@ public static boolean updateBuildConfigFromJob(WorkflowJob job,
-                                    BranchSpec branchSpec = branches.get(0);
-                                    String branch = branchSpec.getName();
-                                    while (branch.startsWith("*")
-                                            || branch.startsWith("/")) {
-                                        branch = branch.substring(1);
-                                    }
-                                    if (!branch.isEmpty()) {
-                                        ref = branch;
-                                    }
-                                }
-                                LOGGER.log(Level.FINE, "updating bc "
-                                        + namespaceName + " gitURL to "
-                                        + gitUrl + " and ref to " + ref);
-                                OpenShiftUtils.updateGitSourceUrl(buildConfig,
-                                        gitUrl, ref);
-                                rc = true;
-                            }
-                        }
-                    }
-                }
-                return rc;
+              SCM scm = cpsScmFlowDefinition.getScm();
+              if (scm instanceof GitSCM) {
+                populateFromGitSCM(buildConfig, source, (GitSCM) scm, null);
+                LOGGER.log(Level.FINE, "updating bc " + namespaceName );
+                rc = true;
+              }
+              return rc;
             }
-            return false;
+          return false;
         }
 
-        if (definition instanceof CpsFlowDefinition) {
-            CpsFlowDefinition cpsFlowDefinition = (CpsFlowDefinition) definition;
-            String jenkinsfile = cpsFlowDefinition.getScript();
-            if (jenkinsfile != null
-                    && jenkinsfile.trim().length() > 0
-                    && !jenkinsfile.equals(jenkinsPipelineStrategy
-                            .getJenkinsfile())) {
-                LOGGER.log(Level.FINE, "updating bc " + namespaceName
-                        + " jenkinsfile to " + jenkinsfile
-                        + " where old jenkinsfile was "
-                        + jenkinsPipelineStrategy.getJenkinsfile());
-                jenkinsPipelineStrategy.setJenkinsfile(jenkinsfile);
-                return true;
+    if (definition instanceof CpsFlowDefinition) {
+      CpsFlowDefinition cpsFlowDefinition = (CpsFlowDefinition) definition;
+      String jenkinsfile = cpsFlowDefinition.getScript();
+      if (jenkinsfile != null && jenkinsfile.trim().length() > 0 && !jenkinsfile.equals(jenkinsPipelineStrategy.getJenkinsfile())) {
+          LOGGER.log(Level.FINE, "updating bc " + namespaceName + " jenkinsfile to " + jenkinsfile + " where old jenkinsfile was " + jenkinsPipelineStrategy.getJenkinsfile());
+        jenkinsPipelineStrategy.setJenkinsfile(jenkinsfile);
+        return true;
+      }
+
+      return false;
+    }
+
+    // support multi-branch or github organization jobs
```

Comment:
shouldn't we be doing some sort of check here to make sure it's the type we expect?

Reply:
If you mean `BranchJobProperty property = job.getProperty(BranchJobProperty.class);` then no, the jenkins API handles that, and only returns something if it is of the requested class.

Reply:
no i mean that above this we're doing various type checks like:
  if (definition instanceof CpsScmFlowDefinition) {
 if (definition instanceof CpsFlowDefinition) {

and then we seem to be doing this as default behavior if neither of those checks are met.  that makes me think we're assuming if this object isn't one of the two we checked for, it will always be some third option we expect.

Reply:
if you're saying we'll always get null back if it's the wrong kind of instance, i guess that works.

Reply:
yes, we should get null back if the wrong kind of instance